metpy/io/pandas_to_netcdf.py

```diff
@@ -0,0 +1,172 @@
+
+    path_to_save : str
+        Path, including filename, for where to save netCDF file.
+
+    netcdf_format : str, optional
+
+    column_units : dict, optional
+        Dictionary of units to attach to columns of the dataframe. Overrides
+        the units attribute if it is attached to the dataframe.
+
+    standard_names : dict, optional
+        Dictionary of variable descriptions that are CF-compliant
+
+    long_names : dict, optional
+        Dictionary of longer variable descriptions that provide more detail
+        than standard_names
+
+    dataset_type: str, optional
+        Type of dataset to be converted. Options are 'timeSeries', 'profile',
+        or 'trajectory'. While optional, this variable should be declared to create
+        a CF-compliant DSG netCDF file.
+
+    Returns
+    -------
+    NetCDF file saved to `path_to_save`.
+
+    """
+    # Verify_integrity must be true in order for conversion to netCDF to work
+    # Return a TypeError if not provided a Pandas DataFrame
+    try:
+        # Create the dimensions for use later in netCDF file
+        samplingindex = df.groupby([sampling_var], sort=False).ngroup()
+        obs = arange(0, len(df))
+        df.insert(0, 'samplingIndex', samplingindex)
+        df.insert(1, 'observations', obs)
+
+        # Handle the sampling location specific data
+        sampling_data = df[sampling_data_vars]
+        samples = sampling_data.groupby([sampling_var], sort=False).ngroup()
+        sampling_data.insert(0, 'samples', samples)
+        sampling_data = sampling_data.groupby('samples').first()
+        dataset_samples = xr.Dataset.from_dataframe(sampling_data)
+
+        # Create the dataset for the variables of each observation
+        df = df.drop(sampling_data_vars, axis=1)
+        df = df.set_index(['observations'], verify_integrity=True)
+        dataset_var = xr.Dataset.from_dataframe(df)
+
+        # Merge the two datasets together
+        dataset_final = xr.merge([dataset_samples, dataset_var], compat='no_conflicts')
+
+    except (AttributeError, ValueError, TypeError):
+        raise TypeError('A pandas dataframe was not provided')
+
+    # Attach variable-specific metadata
+    _assign_metadata(dataset_final, column_units, standard_names, long_names)
+
+    # Attach dataset-specific metadata
+    if dataset_type:
+        dataset_final.attrs['featureType'] = dataset_type
+    else:
+        log.warning('No dataset type provided - netCDF will not have appropriate metadata'
+                    'for a DSG dataset.')
+    if dataset_type:
+        dataset_final[sampling_var].attrs['cf_role'] = dataset_type.lower() + '_id'
+    dataset_final['samplingIndex'].attrs['instance_dimension'] = 'samples'
+
+    # Determine mode to write to netCDF
+    write_mode = 'w'
+    if path.exists(path_to_save):
+        # Eventually switch to 'a' to allow appending and delete error
+        raise ValueError('File already exists - please delete and run again')
```

Comment:
Can probably test this by trying to write to the same file twice. Could also just do a `open(str(tmpdir) + '/test.nc', 'wb')` before calling the save function.